core/src/data/value/mod.rs

```diff
@@ -555,6 +555,45 @@ impl Value {
     pub fn to_cmp_be_bytes(&self) -> Result<Vec<u8>> {
         self.try_into().map(|key: Key| key.to_cmp_be_bytes())
     }
+
+    /// only Value::Str type allowed
+    /// returns positon of substring from string
+    /// if there are substring in string, returns the position where the first letter of the substring begins. The minimum value is 1
+    /// else returns 0 because there is no substring to find within the string
+    /// # Examples
+    /// ```
+    /// use gluesql_core::data::{ValueError, value::Value::*};
+    ///
+    /// let str1 = Str("ramen".to_string());
+    /// let str2 = Str("men".to_string());
+    /// let empty_str = Str("".to_string());
+    /// assert_eq!(str1.str_position(&str2), Ok(3));
+    /// assert_eq!(str2.str_position(&str1), Ok(0));
+    /// assert_eq!(Null.str_position(&str2), Ok(0));
+    /// assert_eq!(str1.str_position(&Null), Ok(0));
+    /// assert_eq!(empty_str.str_position(&str2), Ok(0));
+    /// assert_eq!(str1.str_position(&empty_str), Ok(0));
+    /// assert_eq!(
+    ///     str1.str_position(&I8(1)),
+    ///     Err(ValueError::StrPositionOnNonString(str1, I8(1)).into())
+    /// );
+    /// ```
+    pub fn str_position(&self, other: &Value) -> Result<usize> {
+        use Value::*;
+        match (self, other) {
+            (Str(from_str), Str(sub_str)) => {
+                if from_str == &"".to_string() || sub_str == &"".to_string() {
+                    return Ok(0);
+                }
+                match from_str.find(sub_str).map(|position| position + 1) {
+                    Some(pos) => Ok(pos),
+                    None => Ok(0),
+                }
+            }
+            (Null, Str(_)) | (Str(_), Null) => Ok(0),
+            _ => Err(ValueError::StrPositionOnNonString(self.clone(), other.clone()).into()),
+        }
```

Comment:
```suggestion
        use Value::*;

        match (self, other) {
            (Str(from_str), Str(sub_str)) => {
                if from_str == "" || sub_str == "" {
                    return Ok(0);
                }
                
                Ok(from_str.find(sub_str).map(|position| position + 1).unwrap_or(0))
            }
            (Null, Str(_)) | (Str(_), Null) => Ok(0),
            _ => Err(ValueError::StrPositionOnNonString(self.clone(), other.clone()).into()),
        }
```

Reply:
In mysql, if `Null` is entered, `Null` is output.
Wouldn't it be appropriate to consider this

```sql
mysql> select position(null in null);
+---------------+
| position(null in null) |
+---------------+
| NULL |
+---------------+
1 row in set (0.00 sec)

mysql> select position("" in null);
+-------------------------+
| position("" in null) |
+-------------------------+
| NULL |
+-------------------------+
1 row in set (0.00 sec)

mysql> select position(null in "");
+-------------------------+
| position(null in "") |
+-------------------------+
| NULL |
+-------------------------+
1 row in set (0.00 sec)
```

Reply:
Current return type is Result<usize>.. then, how about Result<Value>.
If we use that we have to consider general case return type. If found, return Result<I64> will be useful